src/cmds/openapi/reduce.ts

```diff
@@ -0,0 +1,189 @@
+              // definitely isn't a tag we want here we need to filter them out.
+              return typeof tag === 'string';
+            });
+
+          return [...new Set(tags)].map(tag => ({
+            title: tag,
+            value: tag,
+          }));
+        },
+      },
+      {
+        type: (prev, values) => (values.reduceBy === 'paths' ? 'multiselect' : null),
+        name: 'paths',
+        message: 'Choose which paths to reduce by:',
+        min: 1,
+        choices: () => {
+          return Object.keys(parsedBundledSpec.paths).map(p => ({
+            title: p,
+            value: p,
+          }));
+        },
+      },
+      {
+        type: (prev, values) => (values.reduceBy === 'paths' ? 'multiselect' : null),
+        name: 'methods',
+        message: 'Choose which HTTP methods that are available across these paths to reduce by:',
+        min: 1,
+        choices: (prev, values) => {
+          const paths: string[] = values.paths;
+          let methods = paths
+            .map((p: string) => Object.keys(parsedBundledSpec.paths[p]))
+            .flat()
+            .filter((method: string) => method.toLowerCase() !== 'parameters');
+
+          methods = [...new Set(methods)];
+          methods.sort();
+
+          return methods.map((method: string) => ({
+            title: method.toUpperCase(),
+            value: method,
+          }));
+        },
+      },
+      {
+        type: 'text',
+        name: 'outputPath',
+        message: 'Enter the path to save your reduced API definition to:',
+        initial: () => {
+          const extension = path.extname(specPath);
+          return `${path.basename(specPath).split(extension)[0]}-reduced${extension}`;
+        },
+        validate: value => {
+          if (value.length) {
+            if (!fs.existsSync(value)) {
+              return true;
+            }
+
+            return 'Specified output path already exists.';
+          }
+
+          return 'An output path must be supplied.';
+        },
+      },
+    ]);
+
+    Command.debug(`reducing by ${promptResults.reduceBy}`);
+    Command.debug(
+      `options being supplied to the reducer: ${JSON.stringify({
+        tags: promptResults.tags,
+        paths: promptResults.tags,
+        methods: promptResults.methods,
+      })}`
+    );
+
+    Command.debug(`about to reduce spec located at ${specPath}`);
+
+    const spinner = ora({ ...oraOptions() });
+    spinner.start('Reducing your API definition...');
+
+    let reducedSpec;
+
+    try {
+      reducedSpec = oasReducer(parsedBundledSpec, {
+        tags: promptResults.tags || [],
+        paths: (promptResults.paths || []).reduce((acc: Record<string, string[]>, p: string) => {
+          acc[p] = promptResults.methods;
+          return acc;
+        }, {}),
+      });
+
+      spinner.succeed(`${spinner.text} done! ✅`);
+    } catch (err) {
+      Command.debug(`err=${err.message}`);
+      throw err;
```

Comment:
The reducer throws exceptions for errors (eg. if you try to filter out too much) right now so instead of trying to capture and put those in some sort of "Yikes! We couldn't reduce this because...` error I'm just letting it do its thing.